src/core/events/bpf.rs

```diff
@@ -109,6 +126,42 @@ impl BpfEventsFactory {
     pub(crate) fn map_fd(&self) -> RawFd {
         self.map.as_fd().as_raw_fd()
     }
+
+    /// Get the log map fd for reuse.
+    pub(crate) fn log_map_fd(&self) -> RawFd {
+        self.log_map.as_fd().as_raw_fd()
+    }
+
+    fn ringbuf_handler<CB>(
+        &self,
+        map: &libbpf_rs::MapHandle,
+        rb_handler: CB,
+    ) -> Result<Option<thread::JoinHandle<()>>>
```

Comment:
I would prefer to not use `Option` here as the function can't return `None`. Also, could you add a small comment above the fct definition?

Reply:
Just to make it clear, are you suggesting:
```
self.handle = Some(self.ringbuf_handler(&self.map, process_event)?);
self.log_handle = Some(self.ringbuf_handler(&self.log_map, process_log)?);
```

`Option` is used for `BpfEventsFactory` (essentially `Default`)